server/routerlicious/packages/lambdas/src/alfred/interfaces.ts

```diff
@@ -56,7 +56,7 @@ export interface IBroadcastSignalEventPayload {
 	/**
 	 * Content of the runtime signal introduced from the broadcast-signal endpoint.
 	 */
-	signalContent: string;
+	signalContent: unknown;
```

Comment:
What is the contract for the data format here? Is it defined by the signal type? These payloads are serializable, right? Should we constrain this type to something that supports serialization?

Reply:
The contract is a stringified version of IRuntimeSignalEnvelope. @alexvy86 why did we change this to unknown again? I can't remember the exact reason -- for a while I was mistakenly using a mixture of JSON and stringified JSON, but the final (correct) version looks like this:

```
curl -X POST -H "Content-Type: application/json" -d '{"signalContent":"{\"contents\":{\"type\":\"ExternalDataChanged_V1.0.0\", \"content\":{\"taskListId\":\"task-list-1\"}}}"}' localhost:7070/tinylicious/76e92769-295e-46d2-8fe9-02aa0ea9447e/broadcast-signal
```

Reply:
Also the route is explicitly throwing if it's not in that type now.

Reply:
Hmm the change to `unknown` was supposedly to avoid the need for the caller of the API to provide it as a string, and instead provide it as an actual JSON object, no?

Reply:
Should this be typed as a `Record`, then?

Reply:
Yeah, that might actually be a better option.

Reply:
We should also be sure to document the expectation that this is a JSONified version of `IRuntimeSignalEnvelope`.

Reply:
Okay, can you take a look at the latest? I still don't quite understand how to use Record -- should I be defining the IRuntimeSignalEnvelope interface as a Record? Also, the Partial<IRuntimeSignalEnvelope> forces me to also make the definition a partial -- is that what we want?

Reply:
Actually, sorry, `Record` may or may not be the best option here. I guess my first question is: should this just be typed as `IRuntimeSignalEnvelope`, if that's the only value we support? Or am I misunderstanding?

If not that, then maybe using our shared `Jsonable` type would be best? It handles recursive cases better than `Record` would.

Reply:
> Also, the Partial forces me to also make the definition a partial -- is that what we want?

`Partial` is a type built into TypeScript. It defines a type that is the same as the parameter you give it, except its members are optional.

E.g. 
```typescript
interface Foo {
  bar: string;
  baz: () => number;
}

/**
 * Equivalent to 
 * {
 *   bar?: string;
 *   baz?: () => number;
 * }
 */
type PartialFoo = Partial<Foo>;
```

Reply:
Got it! Okay so I went ahead and used `IRuntimeSignalEnvelope` here. The only other thing to address I think now is converting it on receipt inside Alfred: https://github.com/microsoft/FluidFramework/pull/17540/files#diff-56222e558ec8d0b289c0be5a65d798a1ec047f1af4ad33aab0c5454e38381f22R148-R151
<img width="878" alt="Screenshot 2023-10-04 at 5 04 49 PM" src="https://github.com/microsoft/FluidFramework/assets/6777404/70efc530-2cb0-4860-aea3-8b198f72da1e">

I would like to prevent using that `as unknown as IRuntimSignalEnvelope` on line 151, but if I remove the `as unknown`  and cast it directly to IRuntimSignalEnvelope, it errors with:

``` Conversion of type 'string' to type 'IRuntimeSignalEnvelope' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.ts(2352)```

Reply:
Ohh I just realized what the problem is -- the function `getParam` requires the second param to be a string, but it isn't anymore, since it's a JSON. Hrm.

Reply:
Okay, I'm grabbing the signalContent directly from the body now, and casting it directly as IRuntimSignalEnvelope. It should all work, please take a look! I wonder how to document now that the body will have to be JSON though :/ ANd do we do a header check?